File: ios/BlueBillywigPlayerViewManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>
#import <React/RCTLog.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(EventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

@end

@interface RCT_EXTERN_MODULE(BlueBillywigPlayerViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(src, NSString)
RCT_EXPORT_VIEW_PROPERTY(autoPlay, BOOL)
RCT_EXPORT_VIEW_PROPERTY(paused, BOOL);
RCT_EXPORT_VIEW_PROPERTY(muted, BOOL);

RCT_EXTERN_METHOD(play:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(pause:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(mute:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(unmute:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(seek:(nonnull NSNumber *)node offsetInSeconds:(nonnull NSNumber *)offset)
// RCT_EXTERN_METHOD(enterFullscreen:(nonnull NSNumber *)node)
// RCT_EXTERN_METHOD(exitFullscreen:(nonnull NSNumber *)node)


@end

File: ios/BlueBillywigPlayerViewManager.swift
import Foundation
import UIKit

@objc(BlueBillywigPlayerViewManager)
class BlueBillywigPlayerViewManager: RCTViewManager {
  private var playerView: BlueBillywigPlayerView?
  private var controller: UIViewController = UIViewController()

  override func view() -> BlueBillywigPlayerView? {
    playerView = BlueBillywigPlayerView(uiViewController: controller)
    controller.view = playerView
    return playerView
  }

  override class func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc func play(_ node: NSNumber) {
    DispatchQueue.main.async {
        let playerView = self.bridge.uiManager.view(
            forReactTag: node
        ) as! BlueBillywigPlayerView
        playerView.play()
    }
  }

  @objc func pause(_ node: NSNumber) {
    DispatchQueue.main.async {
        let playerView = self.bridge.uiManager.view(
            forReactTag: node
        ) as! BlueBillywigPlayerView
        playerView.pause()
    }
  }

  @objc func mute(_ node: NSNumber) {
    DispatchQueue.main.async {
        let playerView = self.bridge.uiManager.view(
            forReactTag: node
        ) as! BlueBillywigPlayerView
        playerView.mute()
    }
  }

  @objc func unmute(_ node: NSNumber) {
    DispatchQueue.main.async {
        let playerView = self.bridge.uiManager.view(
            forReactTag: node
        ) as! BlueBillywigPlayerView
        playerView.unmute()
    }
  }

  @objc func seek(_ node: NSNumber, offsetInSeconds: NSNumber) {
    DispatchQueue.main.async {
        let playerView = self.bridge.uiManager.view(
            forReactTag: node
        ) as! BlueBillywigPlayerView
        playerView.seek(offsetInSeconds: offsetInSeconds)
    }
  }

  // @objc func enterFullscreen(_ node: NSNumber) {
  //   DispatchQueue.main.async {
  //       let playerView = self.bridge.uiManager.view(
  //           forReactTag: node
  //       ) as! BlueBillywigPlayerView
  //       playerView.enterFullscreen()
  //   }
  // }

  // @objc func exitFullscreen(_ node: NSNumber) {
  //   DispatchQueue.main.async {
  //       let playerView = self.bridge.uiManager.view(
  //           forReactTag: node
  //       ) as! BlueBillywigPlayerView
  //       playerView.exitFullscreen()
  //   }
  // }
}

